src/command/pre-release.ts

```diff
@@ -0,0 +1,194 @@
+    let type;
+
+    vorpal
+        .command(`${COMMAND} <branch> [type]`)
+        .description(`Merge <branch>...<head> pull requests across repositories and upsert a release of the specified <type> (valid options: ${RELEASE_TYPES.map((t) => `"${t}"`).join(' or ')}, default: "${RELEASE_TYPES[0]}")`)
+        .validate((input) => {
+            type = input.type || RELEASE_TYPES[0];
+            return RELEASE_TYPES.includes(type) ? true : `Invalid release type "${type}"`;
+        })
+        .action(function action(args) {
+            const { branch } = args;
+            const config = Settings.get<PreReleaseConfiguration>(COMMAND);
+            const isFinalizeStep = (branch === FINALIZE_KEYWORD);
+
+            Settings.getAbsoluteRepositoryPaths(config).forEach((cwd) => {
+                let pr;
+
+                // region 1. Find and merge the pull request.
+                if (!isFinalizeStep) {
+                    this.log(`[${cwd}] Checking if pull request for branch '${branch}' exists`);
+
+                    const prList = Hub.api<IPullRequest[]>(
+                        [`repos/{owner}/{repo}/pulls?head={owner}:${branch}&base=${DEFAULT_BRANCH}`],
+                        cwd,
+                    );
+
+                    if (null === prList.length || 0 === prList.length) {
+                        throw new Error(`Unable to find any open pull requests for branch '${branch}'`);
+                    }
+
+                    [pr] = prList;
+                    this.log(`[${cwd}] Merging pull request ${pr.url}`);
+
+                    Hub.api(
+                        [
+                            `repos/{owner}/{repo}/pulls/${pr.number}/merge`,
+                            '-X',
+                            'PUT',
+                            '-F',
+                            'merge_method=squash',
+                            '-F',
+                            `sha=${pr.head.sha}`,
+                        ],
+                        cwd,
+                    );
+
+                    this.log(`[${cwd}] Merged`);
+                    this.log(`[${cwd}] Deleting branch ${branch}`);
+
+                    Hub.api(
+                        [
+                            `repos/{owner}/{repo}/git/refs/heads/${branch}`,
+                            '-X',
+                            'DELETE',
+                        ],
+                        cwd,
+                    );
+
+                    this.log(`[${cwd}] Deleted`);
+                }
+                // endregion
+
+                // region 2. Determine the target release.
+                const releaseList = Hub.run(
+                    ['release', '-f', '%T', '--exclude-prereleases', '-L', '1'],
+                    cwd,
+                );
+
+                const previousRelease = releaseList || '1.0.0';
+                const semver = new SemVer(previousRelease);
+                const newRelease = semver.inc(type);
```

Comment:
Maybe not a huge deal but something to bring up: The current state would allow you to start a patch and minor release at the same time, which seems a little weird, especially given that the behavior of "drafts" would effectively have both of them pointing at the HEAD of `master`. Perhaps another piece of validation that would be good to get in place?